controllers/reconcile.go

```diff
@@ -590,3 +603,72 @@ func (r *FrontendReconciliation) createSSOConfigMap() (string, error) {
 
 	return hash, nil
 }
+
+func (r *FrontendReconciliation) createServiceMonitor() error {
+
+	// the monitor mode will default to "app-interface"
+	ns := "openshift-customer-monitoring"
+
+	if r.FrontendEnvironment.Spec.Monitoring.Mode == "local" {
+		ns = r.Frontend.Namespace
+	}
+
+	nn := types.NamespacedName{
+		Name:      r.Frontend.Name,
+		Namespace: ns,
+	}
+
+	svcMonitor := &prom.ServiceMonitor{}
+	if err := r.Cache.Create(MetricsServiceMonitor, nn, svcMonitor); err != nil {
+		return err
+	}
+
+	labler := utils.GetCustomLabeler(map[string]string{"prometheus": r.FrontendEnvironment.Name}, nn, r.Frontend)
+	labler(svcMonitor)
+	svcMonitor.SetOwnerReferences([]metav1.OwnerReference{r.Frontend.MakeOwnerReference()})
+
+	svcMonitor.Spec.Endpoints = []prom.Endpoint{{
+		Path:     "/metrics",
+		Port:     "metrics",
+		Interval: prom.Duration("15s"),
+	}}
+	svcMonitor.Spec.NamespaceSelector = prom.NamespaceSelector{
+		MatchNames: []string{r.Frontend.Namespace},
+	}
+	svcMonitor.Spec.Selector = metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"frontend": r.Frontend.Name,
+		},
+	}
+
+	if err := r.Cache.Update(MetricsServiceMonitor, svcMonitor); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *FrontendReconciliation) removeExistingServiceMonitor() error {
+	ns := "openshift-customer-monitoring"
```

Comment:
This code is also in the above function - consider moving out into its own function as hardcoded names are nicer in one place. Also consider making the hardcoded name a const.